src/Native/Unix/Common/pal_compiler.h

```diff
@@ -20,3 +20,5 @@
 #else
 #define c_static_assert(e) typedef char __c_static_assert__[(e)?1:-1]
 #endif
+
+#define DLLEXPORT __attribute__ ((__visibility__ ("default")))
```

Comment:
Change the default visibility for System.Native to hidden to ensure that all exports are marked with this? (It should also make it a bit smaller.)

`add_compile_options(-fvisibility=hidden)`

Reply:
It would force the use of `DLLEXPORT` for C++ files as well. That's definitely something I could do, so it's mostly up to you. Also, it would require me to make sure all `SystemNative_*` functions are declared in the header properly.

Reply:
> It would force the use of DLLEXPORT for C++ files as well.

I think that it is fine. It would ensure that any new additions do have the `DLLEXPORT` attribute - reduce the chance of breaking Mono flavor of System.Native.

Also, I hope that all `System.Native` get converted to C in not too distant future. We are 75% there with this PR (~165kB converted, 55kB remaining).

Reply:
Fixed

Reply:
> I hope that all `System.Native` get converted to C in not oo distant future.

👍  almost more than half way:

![image](https://user-images.githubusercontent.com/33230602/39427608-17a71bd0-4c8d-11e8-8394-de7e03c7ff33.png)